src/qrengine.c

```diff
@@ -185,4 +218,47 @@ int qrgen_string_to_qr_pngfile(const char *filename, const char* inString, uint8
         printf("Error generating QR\n");
         return 1;
     }
+}
+
+
+// encode a string to JPEG file with med ECC
+int qrgen_string_to_qr_jpgfile(const char* outFilename, const char* inString, uint8_t sizeMultiplier)
+{
+    if (sizeMultiplier < 1) {
+        sizeMultiplier = 1;
+    }
+
+    unsigned char* jpg;
+    int jpgsize;
+    unsigned width;
+    unsigned height;
+
+    uint8_t tempBuffer[qrcodegen_BUFFER_LEN_MAX];
+    uint8_t qrcode[qrcodegen_BUFFER_LEN_MAX];
+
+    bool codeGenerated = qrcodegen_encodeText(inString, tempBuffer, qrcode, qrcodegen_Ecc_MEDIUM, qrcodegen_VERSION_MIN, qrcodegen_VERSION_MAX, qrcodegen_Mask_AUTO, true);
+
+    if (codeGenerated) {
+        unsigned tempSideDim = qrcodegen_getSize(qrcode);
+        width = tempSideDim * sizeMultiplier;
+        height = tempSideDim * sizeMultiplier;
+
+        // bytes to monobitmap with sizemodifier
+        uint8_t* image = bytesToMono(qrcode, sizeMultiplier);
+
+        //encode jpeg
+        jpec_enc_t* enc = jpec_enc_new(image, width, height);
+        const uint8_t* jpeg = jpec_enc_run(enc, &jpgsize);
+
+        FILE* file = fopen(outFilename, "wb");
+        fwrite(jpeg, sizeof(uint8_t), jpgsize, file);
+        fclose(file);
+        jpec_enc_del(enc);
+        free(image);
+    } 
+    else 
+    {
+        printf("Error generating QR\n");
+        return 1;
+    }
 }
```

Comment:
New line. Also perhaps consider as 1 is truthy and the enclosure of codeGenerated doesn't return anything:
```

// encode a string to JPEG file with med ECC
int qrgen_string_to_qr_jpgfile(const char* outFilename, const char* inString, uint8_t sizeMultiplier)
{
    if (sizeMultiplier < 1) {
        sizeMultiplier = 1;
    }

    // unsigned char* jpg;
    int jpgsize;
    unsigned width;
    unsigned height;

    uint8_t tempBuffer[qrcodegen_BUFFER_LEN_MAX];
    uint8_t qrcode[qrcodegen_BUFFER_LEN_MAX];

    bool codeGenerated = qrcodegen_encodeText(inString, tempBuffer, qrcode, qrcodegen_Ecc_MEDIUM, qrcodegen_VERSION_MIN, qrcodegen_VERSION_MAX, qrcodegen_Mask_AUTO, true);

    if (codeGenerated) {
        unsigned tempSideDim = qrcodegen_getSize(qrcode);
        width = tempSideDim * sizeMultiplier;
        height = tempSideDim * sizeMultiplier;

        // bytes to monobitmap with sizemodifier
        uint8_t* image = bytesToMono(qrcode, sizeMultiplier);

        //encode jpeg
        jpec_enc_t* enc = jpec_enc_new(image, width, height);
        const uint8_t* jpeg = jpec_enc_run(enc, &jpgsize);

        FILE* file = fopen(outFilename, "wb");
        fwrite(jpeg, sizeof(uint8_t), jpgsize, file);
        fclose(file);
        jpec_enc_del(enc);
        free(image);
    } 
    else 
    {
        printf("Error generating QR\n");
        return 0;
    }
    return codeGenerated;
}

```

Reply:
think i got it all now.